crates/resource-resolver/src/lib.rs

```diff
@@ -0,0 +1,192 @@
+/// **The feature prevents the creation of hardlinks that the user does not own
+/// or have read/write access to.** [See the patch that enabled this].
+///
+/// [Hard Link]: https://en.wikipedia.org/wiki/Hard_link
+/// [See the patch that enabled this]: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=800179c9b8a1e796e441674776d11cd4c05d61d7
+pub fn current_exe() -> Result<PathBuf> {
+    starting_binary::STARTING_BINARY
+        .cloned()
+        .map_err(|e| Error::Io("Can't detect the path of the current exe".to_string(), e))
+}
+
+/// Retreive the resource path of your app, packaged with cargo packager.
+/// This function behave the same as [`resource_dir`], except it accepts
+/// a parameter that will be happened to the resource path when no packaging format
+/// is used.
+///
+/// Example: You want to include the folder `crate/resource/icons/`.
+///
+/// - With `cargo run` command, you will have to execute
+///     `resource_dir().unwrap().join("resource/icons/")` to get the path.
+/// - With any other formats, it will be `resource_dir().unwrap().join("icons/")`.
+///
+/// ```
+/// use cargo_packager_resource_resolver as resource_resolver;
+/// use resource_resolver::{PackageFormat, resource_dir_with_suffix};
+///
+/// resource_dir_with_suffix(PackageFormat::None, "resource").unwrap().join("icons/");
+/// ```
+pub fn resource_dir_with_suffix(package_format: PackageFormat, suffix: &str) -> Result<PathBuf> {
+    match package_format {
+        PackageFormat::None => {
+            let root_crate_dir = env::var("CARGO_MANIFEST_DIR")
+                .map_err(|e| {
+                    match e {
+                        env::VarError::NotPresent => {
+                            Error::Env("PackageFormat::None was use, but CARGO_MANIFEST_DIR environnement variable was not defined".to_string())
+                        },
+                        _ => Error::Var("Can't access CARGO_MANIFEST_DIR environnement variable".to_string(), e)
+                    }
+                })?;
+            Ok(PathBuf::from(root_crate_dir).join(suffix))
+        }
+        PackageFormat::App | PackageFormat::Dmg => {
+            let exe = current_exe()?;
+            let exe_dir = exe.parent().unwrap();
+            exe_dir
+                .join("../Resources")
+                .canonicalize()
+                .map_err(|e| Error::Io("".to_string(), e))
+        }
+        PackageFormat::Wix => Err(Error::UnsupportedPlatform),
+        PackageFormat::Nsis => {
+            let exe = current_exe()?;
+            let exe_dir = exe.parent().unwrap();
+            Ok(exe_dir.to_path_buf())
+        }
+        PackageFormat::Deb => {
+            // maybe this is not reliable, and we need to get the app name from argument
+            let exe = current_exe()?;
+            let binary_name = exe.file_name().unwrap().to_string_lossy();
+
+            let path = format!("/usr/lib/{}/", binary_name);
```

Comment:
@lucasfernog-crabnebula what do you think about this?

Reply:
this is the Tauri impl FYI: https://github.com/tauri-apps/tauri/blob/97e334129956159bbd60e1c531b6acd3bc6139a6/core/tauri-utils/src/platform.rs#L226
it's ugly but it's more tested right now

Reply:
we should also apply that impl to the AppImage format